cli/src/main/java/de/jplag/cli/server/ReportViewer.java

```diff
@@ -0,0 +1,107 @@
+package de.jplag.cli.server;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+
+/**
+ * Manages the internal report viewer. Serves the static files for the report viewer and the results.zip.
+ */
+public class ReportViewer implements HttpHandler {
+    private static final Logger logger = LoggerFactory.getLogger(ReportViewer.class);
+    private static final int SUCCESS_RESPONSE = 200;
+    private static final int NOT_FOUND_RESPONSE = 404;
+
+    private final RoutingTree routingTree;
+
+    private HttpServer server;
+
+    /**
+     * @param zipFile The zip file to use for the report viewer
+     * @throws IOException If the zip file cannot be read
+     */
+    public ReportViewer(File zipFile) throws IOException {
+        this.routingTree = new RoutingTree();
+
+        this.routingTree.insertRouting("", new RoutingResources("report-viewer").or(new RoutingAlias("index.html")));
+        this.routingTree.insertRouting("results.zip", new RoutingStaticFile(zipFile, ContentType.ZIP));
+    }
+
+    /**
+     * Starts the server
+     * @return The port the server runs at
+     * @throws IOException If the server cannot be started
+     */
+    public int start() throws IOException {
+        if (server != null) {
+            throw new IllegalStateException("Server already started");
+        }
+        server = HttpServer.create(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0), 0);
```

Comment:
If I understood it correctly, at the moment we let the system pick a port. This will make sharing links with other people (was a requested feature) nearly impossible. We should either pick a default port or let the user set a port via a flag

Reply:
I'd agree here. You could simply make it configurable. Add a default port; e.g., 1996 (Year of JPlag :D)

Reply:
+1 for 1996 :D

Reply:
What I meant here, was sharing as in sending somebody the zip and the link to the comparison. This was shortly discussed in the workshop last summer.
For this a default/setable port is needed, so that the link works without changing the port